### ISSUE TITLE
Ability to monitor XDP handled packets

### ISSUE THREAD

**Issue by oblazek** [cilium/hubble#1339]
Seems like hubble (nor cilium monitor) cannot observe traffic to services which are XDP accelerated. We've been using standalone L4LB in production for approx 2 years now, but haven't used hubble so far in this context.

**Comment by gandro** [comment]
Thanks for raising this. These errors really should be investigated, I suspect this is because L4LB does IP-in-IP encapsulation. Potentiality related issue https://github.com/cilium/cilium/issues/15515

**Comment by oblazek** [comment]
we might have a solution for this.. my colleague will try to send a PR with the fix, once I verify that it works

**Comment by oblazek** [comment]
yeah looks like we can fix the `failed to decode payload`, but still can't monitor xdp handled traffic :/

**Comment by gandro** [comment]
What do you see in `cilium monitor`? Do you observe usable events there? Otherwise we have to investigate what exactly the datapath sends down to userspace in XDP mode.

**Comment by oblazek** [comment]
unfortunately there seem to be **nothing** that is handled by xdp :/

**Comment by oblazek** [comment]
I can investigate myself but would be glad for some pointers

**Comment by gandro** [comment]
@borkmann Do you know if/where we emit trace events from the XDP datapath (the monitor ones, not the pcap recorder)? Or was this never implemented?

**Comment by oblazek** [comment]
imho this is the issue: https://github.com/cilium/cilium/issues/19703

**Comment by brb** [comment]
MBOI https://github.com/cilium/pwru/pull/339.

**Comment by oblazek** [comment]
let's see what comes up from tomorrows summit 🙂

**Comment by brb** [comment]
`pwru` with the XDP tracing is finally released - https://github.com/cilium/pwru/releases/tag/v1.0.7.
